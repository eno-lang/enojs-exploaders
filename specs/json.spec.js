const eno = require('enojs');
const loaders = require('../loaders.js');

describe('json', () => {

  let document;

  describe('{ "valid": true }', () => {
    it('returns { "valid": true }', () => {
      document = eno.parse('json: { "valid": true }');
      expect(document.field('json', loaders.json)).toEqual({ valid: true });
    });
  });

  describe('42', () => {
    it('returns 42', () => {
      document = eno.parse('json: 42');
      expect(document.field('json', loaders.json)).toBe(42);
    });
  });

  describe('["valid", true]', () => {
    it('returns ["valid", true]', () => {
      document = eno.parse('json: ["valid", true]');
      expect(document.field('json', loaders.json)).toEqual(['valid', true]);
    });
  });

  describe('invalid', () => {
    it('throws an error', () => {
      document = eno.parse('json: invalid');
      expect(() => document.field('json', loaders.json)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('{ invalid: true }', () => {
    it('throws an error', () => {
      document = eno.parse('json: { invalid: true }');
      expect(() => document.field('json', loaders.json)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('{ "invalid": true, }', () => {
    it('throws an error', () => {
      document = eno.parse('json: { "invalid": true, }');
      expect(() => document.field('json', loaders.json)).toThrowErrorMatchingSnapshot();
    });
  });
});
