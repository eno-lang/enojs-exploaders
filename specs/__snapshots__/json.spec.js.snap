// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json { "invalid": true, } throws an error 1`] = `
"'json' contains invalid json: Unexpected token } in JSON at position 19

   Line | Content
 >    1 | json: { \\"invalid\\": true, }
"
`;

exports[`json { invalid: true } throws an error 1`] = `
"'json' contains invalid json: Unexpected token i in JSON at position 2

   Line | Content
 >    1 | json: { invalid: true }
"
`;

exports[`json invalid throws an error 1`] = `
"'json' contains invalid json: Unexpected token i in JSON at position 0

   Line | Content
 >    1 | json: invalid
"
`;
